src/array/cpu/spmat_op_impl_coo.cc

```diff
@@ -339,8 +340,9 @@ CSRMatrix COOToCSR(COOMatrix coo) {
     ret_indices = coo.col;
     ret_data = coo.data;
   } else {
+    static_cast<IdType*>(ret_indptr->data)[0] = 0;
     // compute indptr
-    IdType* Bp = static_cast<IdType*>(ret_indptr->data);
+    IdType* Bp = static_cast<IdType*>(ret_indptr_view->data);
```

Comment:
We could further simplify it as:
```c++
} else {
  IdType* Bp = static_cast<IdType*>(ret_indptr->data);
  *(Bp++) = 0;
  ...  // the rest part is the same as your change
}
```
It seems that there is no need to create a view at all.

Reply:
I followed the suggestion. Thank you.